libraries/chain/db_maint.cpp

```diff
@@ -800,6 +800,61 @@ void update_and_match_call_orders( database& db )
    wlog( "Done updating all call orders for hardfork core-343 at block ${n}", ("n",db.head_block_num()) );
 }
 
+/******
+ * @brief cleanup feeds with wrong assets
+ *
+ * Prior to hardfork 868, switching a bitasset's shorting asset would not reset its
+ * feeds. This method will run at the hardfork time, and erase (or nullify) feeds
+ * that have incorrect backing assets.
+ *
+ * @param db the database
+ */
+void cleanup_invalid_feeds_hf_868( database& db )
+{
+   // for each market issued asset
+   const auto& asset_idx = db.get_index_type<asset_index>().indices().get<by_type>();
+   for(auto asset_itr = asset_idx.lower_bound(true); asset_itr != asset_idx.end(); ++asset_itr)
+   {
+      const auto& asset = *asset_itr;
+      // Incorrect witness & committee feeds can simply be removed.
+      // For non-witness-fed and non-committee-fed assets, set incorrect
+      // feeds to price(), since we can't simply remove them. For more information:
+      // https://github.com/bitshares/bitshares-core/pull/832#issuecomment-384112633
+      bool is_witness_or_committee_fed = false;
+      if ( asset.options.flags & (witness_fed_asset | committee_fed_asset) )
+         is_witness_or_committee_fed = true;
+
+      // for each feed
+      const asset_bitasset_data_object& bitasset_data = asset.bitasset_data(db);
+      auto itr = bitasset_data.feeds.begin();
+      while (itr != bitasset_data.feeds.end())
+      {
+         // If the feed is invalid
+         if ( (*itr).second.second.settlement_price.quote.asset_id != bitasset_data.options.short_backing_asset) {
+            db.modify(bitasset_data, [&itr, is_witness_or_committee_fed](asset_bitasset_data_object& obj) {
```

Comment:
Perhaps it's better to move the `while` loop into `db.modify` lambda? Since we're updating the same object, it's better to update only once.

Reply:
That would depend on how "heavyweight" a call to db.modify is. And you would know better than I. If I put the while loop inside, db.modify will always be called for each bitasset, regardless of if it has invalid feeds. Is that what we want?

Reply:
You're correct. Actually in production there won't be many assets need to be updated (if any). Perhaps better add some logging code so we'll know what assets have been updated due to incorrect data.

Reply:
Done